### PR TITLE
Issue 18010 redirect to login

### PR DESCRIPTION
**Pull request by wezell**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 1]
```diff
@@ -45,48 +46,67 @@ protected final void service(HttpServletRequest req, HttpServletResponse respons
...
-        } else {
+            return;
+        } 
 
-            if ((DbConnectionFactory.isMsSql() && LicenseUtil.getLevel() < LicenseLevel.PROFESSIONAL.level) ||
-                    (DbConnectionFactory.isOracle() && LicenseUtil.getLevel() < LicenseLevel.PRIME.level) ||
-                    (!LicenseUtil.isASAllowed())) {
-                Logger.error(this, "Enterprise License is required");
+        if ((DbConnectionFactory.isMsSql() && LicenseUtil.getLevel() < LicenseLevel.PROFESSIONAL.level)
+                        || (DbConnectionFactory.isOracle() && LicenseUtil.getLevel() < LicenseLevel.PRIME.level)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 2]
```diff
@@ -45,48 +46,67 @@ protected final void service(HttpServletRequest req, HttpServletResponse respons
...
+                        || (!LicenseUtil.isASAllowed())) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 3]
```diff
@@ -35,58 +38,82 @@
...
-
-            if ((DbConnectionFactory.isMsSql() && LicenseUtil.getLevel() < LicenseLevel.PROFESSIONAL.level) ||
-                    (DbConnectionFactory.isOracle() && LicenseUtil.getLevel() < LicenseLevel.PRIME.level) ||
-                    (!LicenseUtil.isASAllowed())) {
-                Logger.error(this, "Enterprise License is required");
+            return;
+        } 
+        
+        // if you are not running ee
+        if ((DbConnectionFactory.isMsSql() && LicenseUtil.getLevel() < LicenseLevel.PROFESSIONAL.level)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 4]
```diff
@@ -35,58 +38,82 @@
...
     @CloseDB
     protected final void service(HttpServletRequest req, HttpServletResponse response) throws ServletException, IOException {
-        VelocityRequestWrapper request = new VelocityRequestWrapper(req);
+        final VelocityRequestWrapper request = new VelocityRequestWrapper(req);
         final String uri = CMSUrlUtil.getCurrentURI(request);
-
+        final boolean comeFromSomeWhere = request.getHeader("referer") != null;
+        final User frontEndUser = Try.of(()->WebAPILocator.getUserWebAPI().getLoggedInFrontendUser(request)).getOrNull();
+        request.setRequestUri(uri);
+        final PageMode mode = PageMode.getWithNavigateMode(request);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 5]
```diff
@@ -35,58 +38,82 @@
     @Override
     @CloseDB
     protected final void service(HttpServletRequest req, HttpServletResponse response) throws ServletException, IOException {
-        VelocityRequestWrapper request = new VelocityRequestWrapper(req);
+        final VelocityRequestWrapper request = new VelocityRequestWrapper(req);
         final String uri = CMSUrlUtil.getCurrentURI(request);
-
+        final boolean comeFromSomeWhere = request.getHeader("referer") != null;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 6]
```diff
@@ -35,58 +38,82 @@
...
-            if ((DbConnectionFactory.isMsSql() && LicenseUtil.getLevel() < LicenseLevel.PROFESSIONAL.level) ||
-                    (DbConnectionFactory.isOracle() && LicenseUtil.getLevel() < LicenseLevel.PRIME.level) ||
-                    (!LicenseUtil.isASAllowed())) {
-                Logger.error(this, "Enterprise License is required");
+            return;
+        } 
+        
+        // if you are not running ee
+        if ((DbConnectionFactory.isMsSql() && LicenseUtil.getLevel() < LicenseLevel.PROFESSIONAL.level)
+                        || (DbConnectionFactory.isOracle() && LicenseUtil.getLevel() < LicenseLevel.PRIME.level)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 7]
```diff
@@ -35,58 +38,82 @@
...
+                        || (!LicenseUtil.isASAllowed())) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 8]
```diff
@@ -35,58 +38,82 @@
     @Override
     @CloseDB
     protected final void service(HttpServletRequest req, HttpServletResponse response) throws ServletException, IOException {
-        VelocityRequestWrapper request = new VelocityRequestWrapper(req);
+        final VelocityRequestWrapper request = new VelocityRequestWrapper(req);
         final String uri = CMSUrlUtil.getCurrentURI(request);
-
+        final boolean comeFromSomeWhere = request.getHeader("referer") != null;
+        final User frontEndUser = Try.of(()->WebAPILocator.getUserWebAPI().getLoggedInFrontendUser(request)).getOrNull();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 9]
```diff
@@ -35,58 +37,82 @@
     @Override
     @CloseDB
     protected final void service(HttpServletRequest req, HttpServletResponse response) throws ServletException, IOException {
-        VelocityRequestWrapper request = new VelocityRequestWrapper(req);
+        final VelocityRequestWrapper request = new VelocityRequestWrapper(req);
         final String uri = CMSUrlUtil.getCurrentURI(request);
-
+        final boolean comeFromSomeWhere = request.getHeader("referer") != null;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 10]
```diff
@@ -35,58 +37,82 @@
...
-                    (DbConnectionFactory.isOracle() && LicenseUtil.getLevel() < LicenseLevel.PRIME.level) ||
-                    (!LicenseUtil.isASAllowed())) {
-                Logger.error(this, "Enterprise License is required");
+            return;
+        } 
+        
+        // if you are not running ee
+        if ((DbConnectionFactory.isMsSql() && LicenseUtil.getLevel() < LicenseLevel.PROFESSIONAL.level)
+                        || (DbConnectionFactory.isOracle() && LicenseUtil.getLevel() < LicenseLevel.PRIME.level)
+                        || (!LicenseUtil.isASAllowed())) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 11]
```diff
@@ -35,58 +37,82 @@
...
     @CloseDB
     protected final void service(HttpServletRequest req, HttpServletResponse response) throws ServletException, IOException {
-        VelocityRequestWrapper request = new VelocityRequestWrapper(req);
+        final VelocityRequestWrapper request = new VelocityRequestWrapper(req);
         final String uri = CMSUrlUtil.getCurrentURI(request);
-
+        final boolean comeFromSomeWhere = request.getHeader("referer") != null;
+        final User frontEndUser = ((UserWebAPIImpl) WebAPILocator.getUserWebAPI()).getLoggedInFrontendUser(request);
+        request.setRequestUri(uri);
+        final PageMode mode = PageMode.getWithNavigateMode(request);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 12]
```diff
@@ -35,58 +37,82 @@
...
-            if ((DbConnectionFactory.isMsSql() && LicenseUtil.getLevel() < LicenseLevel.PROFESSIONAL.level) ||
-                    (DbConnectionFactory.isOracle() && LicenseUtil.getLevel() < LicenseLevel.PRIME.level) ||
-                    (!LicenseUtil.isASAllowed())) {
-                Logger.error(this, "Enterprise License is required");
+            return;
+        } 
+        
+        // if you are not running ee
+        if ((DbConnectionFactory.isMsSql() && LicenseUtil.getLevel() < LicenseLevel.PROFESSIONAL.level)
+                        || (DbConnectionFactory.isOracle() && LicenseUtil.getLevel() < LicenseLevel.PRIME.level)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 13]
```diff
@@ -35,58 +37,82 @@
     @Override
     @CloseDB
     protected final void service(HttpServletRequest req, HttpServletResponse response) throws ServletException, IOException {
-        VelocityRequestWrapper request = new VelocityRequestWrapper(req);
+        final VelocityRequestWrapper request = new VelocityRequestWrapper(req);
         final String uri = CMSUrlUtil.getCurrentURI(request);
-
+        final boolean comeFromSomeWhere = request.getHeader("referer") != null;
+        final User frontEndUser = ((UserWebAPIImpl) WebAPILocator.getUserWebAPI()).getLoggedInFrontendUser(request);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 14]
```diff
@@ -35,58 +37,82 @@
...
-
-            if ((DbConnectionFactory.isMsSql() && LicenseUtil.getLevel() < LicenseLevel.PROFESSIONAL.level) ||
-                    (DbConnectionFactory.isOracle() && LicenseUtil.getLevel() < LicenseLevel.PRIME.level) ||
-                    (!LicenseUtil.isASAllowed())) {
-                Logger.error(this, "Enterprise License is required");
+            return;
+        } 
+        
+        // if you are not running ee
+        if ((DbConnectionFactory.isMsSql() && LicenseUtil.getLevel() < LicenseLevel.PROFESSIONAL.level)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)

**Review by jgambarios — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**